packages/plugin-ext/src/hosted/browser/hosted-plugin.ts

```diff
@@ -271,6 +280,55 @@ export class HostedPluginSupport {
         await Promise.all(promises);
     }
 
+    protected async activateByWorkspaceContains(manager: PluginManagerExt, plugin: PluginMetadata): Promise<void> {
+        if (!plugin.source.activationEvents) {
+            return;
+        }
+        const paths: string[] = [];
+        const includePatterns: string[] = [];
+        for (const activationEvent of plugin.source.activationEvents) {
+            if (activationEvent.startsWith('workspaceContains:')) {
+                const fileNameOrGlob = activationEvent.substr('workspaceContains:'.length);
+                if (fileNameOrGlob.indexOf('*') >= 0 || fileNameOrGlob.indexOf('?') >= 0) {
```

Comment:
I've aligned it with how VS Code works.

Reply:
I see. Can you please fully align with VS Code?
```ts
if (/^workspaceContains:/.test(activationEvent)) {
```

Out of curiosity, do not we need a CQ for this? The code is almost the same as in https://github.com/microsoft/vscode/blob/da5fb7d5b865aa522abc7e82c10b746834b98639/src/vs/workbench/api/node/extHostExtensionService.ts#L457-L469

Reply:
`/^workspaceContains:/.test(activationEvent)` what is the difference to `startWith`? except that it removes unnecessary regexp?

let's open a CQ, just to be safe, it would be good to track all CQs i have feelings that we open CQs over and over for already covered code

Reply:
> i have feelings that we open CQs over and over for already covered code

I am with you! On the other hand, I want to avoid any further delays with the Eclipse move just because we forgot to submit a CQ for something.

Reply:
https://dev.eclipse.org/ipzilla/show_bug.cgi?id=20357